test/extended/openstack/cpms.go

```diff
@@ -109,17 +126,30 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] ControlPlane Mach
 				subnetName := subnet.Get("filter.name").String()
 				machineSubnets[subnetName] = struct{}{}
 			}
-			if machineNovaAvailabilityZone != "" {
-				machineNovaAvailabilityZones = append(machineNovaAvailabilityZones, machineNovaAvailabilityZone)
+			if machineNovaAz != "" || machineCinderAz != "" || machineVolumeType != "" {
```

Comment:
You don't need these `!= ""` checks, as the struct properties will be the empty string anyway

Reply:
Sorted out out-of-band: I now think we need this check